### DISCUSSION TITLE
Is StrictMode still necessary?

### DISCUSSION THREAD

**Discussion by raibima** [reactwg/react-compiler#28] [General Questions] [answered]
With the new eslint rules, wondering if the runtime `<StrictMode>` component is still necessary?

**Comment by guillaumebrunerie** [comment, marked as answer]
Yes, for instance if you have an effect that doesn't clean up after itself properly, this will be highlighted by `<StrictMode>` (by breaking your app), but it isn't something a compiler can detect.
